wstool/MemoryTool.cpp

```diff
@@ -0,0 +1,261 @@
+                {
+                    if (InputVirAddr[key].size() > 2)
+                    {
+                        if (InputVirAddr[key][2] > time)
+                        {
+                            break;
+                        }
+                    }
+
+                    if (time > InputVirAddr[key][InputVirAddr[key].size() - 1])
+                    {
+                        inputList.push_back(key);
+                        InputVirAddr.erase(key);
+                        InputkeyVector.erase(itv);
+                    }
+                }
+            }
+            alivenumber = alivenumber - inputList.size();
+            if (maxInput < alivenumber)
+            {
+                maxInput = alivenumber;
+            }
+            if (writeAll)
+            {
+                inputWS.push_back(alivenumber);
+            }
+            if (time % 1000 == 0)
+            {
+                float ProcessRatio = (float)time / WorkingSet::timing;
+                printf("process time :%ld, %ld, %.5f \n", time, WorkingSet::timing, ProcessRatio);
+            }
+        }
+    }
+    else
+    {
+        for (auto &key : InternelkeyVector)
+        {
+            if (InternelVirAddr[key].size() < 3)
+            {
+                maxOutput++;
+            }
+        }
+        for (int64_t time = 0; time < internalSampleTime; time++)
+        {
+            vector<string> timeline;
+            for (auto itv = InternelSampleKeyVector.begin(); itv != InternelSampleKeyVector.end(); ++itv)
+            {
+                string key = *itv;
+                if (!InternelSampleVirAddr[key].empty())
+                {
+
+                    if (InternelSampleVirAddr[key][1] > time)
+                    {
+                        break;
+                    }
+                    if (InternelSampleVirAddr[key].size() > 2)
+                    {
+                        if (InternelSampleVirAddr[key][1] <= time && InternelSampleVirAddr[key][InternelSampleVirAddr[key].size() - 1] > time)
+                        {
+                            timeline.push_back(key);
+                        }
+                        else if (time > InternelSampleVirAddr[key][InternelSampleVirAddr[key].size() - 1])
+                        {
+                            InternelSampleVirAddr.erase(key);
+                            InternelSampleKeyVector.erase(itv);
+                        }
+                    }
+                    else
+                    {
+                        InternelSampleVirAddr.erase(key);
+                        InternelSampleKeyVector.erase(itv);
+                    }
+                }
+            }
+            if (maxinternal < timeline.size())
+            {
+                maxinternal = timeline.size();
+            }
+        }
+    }
+    maxInput = alivenumber;
+    printf("maxInput: %lu \n maxinternal: %lu \n maxOutput: %lu \n", maxInput, maxinternal, maxOutput);
```

Comment:
Another print that has to go. If you need this info it should be in the output file.